extensions/ql-vscode/src/CodeLensProvider.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  CodeLensProvider,
+  TextDocument,
+  CodeLens,
+  Command,
+  Range
+} from 'vscode';
+
+class MyCodeLensProvider implements CodeLensProvider {
+  // Each provider requires a provideCodeLenses function which will give the various documents
+  // the code lenses
+  async provideCodeLenses(document: TextDocument): Promise<CodeLens[]> {
+
+    const codeLenses: CodeLens[] = [];
+
+    for (let index = 0; index < document.lineCount; index++) {
+      const textLine = document.lineAt(index);
+      const regex = new RegExp(/(?=(\w+\(\s*.*(?:,\s*)*\)\s*\{))\1/g);
```

Comment:
Making this change. I've always struggled writing regex (although I'm getting more comfortable with it now), so I always appreciate feedback. The only reason I included the positive lookahead was I was getting a code scanning failure about potential exponential backtracking. But I have also since simplified the regex, so maybe that won't be an issue this time.

Reply:
Yup I think your new regex is free of this problem, because of the simpler handling of the `(...)` parameters!

If you'd like to learn more about exponential regexes, the CodeQL docs for the alert you saw are pretty good: https://codeql.github.com/codeql-query-help/javascript/js-redos/. Exponential backtracking usually happens if you have something like `(pattern1|pattern2)*`, because at each step of matching the next character in the `*` sequence, we don't know whether we just saw a `pattern1` or a `pattern2`, and have to backtrack to try both options if the next match fails.